nixos/modules/services/web-apps/nextcloud-backup.nix

```diff
@@ -0,0 +1,196 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.nextcloud;
+  backupRsyncCommand = "${lib.getExe pkgs.rsync} -rlt --del --perms";
+  restoreRsyncCommand = "${backupRsyncCommand} --chmod=u+rwX --chown=nextcloud:nextcloud";
+  dbBackupRestore =
+    if cfg.config.dbtype == "sqlite" then
+      {
+        # The sqlite database is automatically backed up since it's contained in the datadir
```

Comment:
I think even though the database is contained int the datadir you should still export it as suggested in https://docs.nextcloud.com/server/latest/admin_manual/maintenance/backup.html. I'm not sure maintenance mode guarantees the database will not be touched. Additionally sqlites usage of a WAL might make the backup inconsistent.